packages/core/src/database/find-many.test.ts

```diff
@@ -0,0 +1,55 @@
+import { Users } from '@logto/schemas';
+
+import { createTestPool } from '@/utils/test-utils';
+
+import { buildFindMany } from './find-many';
+
+describe('buildFindMany()', () => {
+  it('matches expected sql', async () => {
+    const pool = createTestPool(
+      'select "id", "username", "primary_email", "primary_phone", "password_encrypted", "password_encryption_method", "password_encryption_salt"\nfrom "users"'
+    );
+    const findMany = buildFindMany(pool, Users);
+    await findMany();
+  });
+
+  it('matches expected sql with where conditions', async () => {
+    const pool = createTestPool(
+      'select "id", "username", "primary_email", "primary_phone", "password_encrypted", "password_encryption_method", "password_encryption_salt"\nfrom "users"\nwhere "id"=$1'
+    );
+    const findMany = buildFindMany(pool, Users);
+    await findMany({ where: { id: '123' } });
```

Comment:
Feels not able to test the findMany returned query script in this way.  
Could mock the returning value using `createTestPool` and test the returning value. Of expect not to throw any exceptions?

Reply:
We may come up with a better query test practice later.

Reply:
discussed offline, `buildFindMany` created a deferred function that will be invoked in `findMany`